### PR TITLE
add third party licenses

### PR DESCRIPTION
**Pull request by Choraden**
The licenses are generated with go-licenses tool.
The template is declared in `third_party_license.tpl`

### CODE REVIEW

**Comment by Choraden** [comment]
@mmatczuk how do you like the format? For me, this is a little too robust. 

Do we want to automate updating these licenses?

File: Makefile [hunk 1]
```diff
@@ -14,6 +14,7 @@ install-dependencies:
 	go install github.com/goreleaser/goreleaser@$(GORELEASER_VERSION)
 	go install golang.org/x/tools/cmd/godoc@latest
 	go install golang.org/x/tools/cmd/stringer@latest
+	go install github.com/google/go-licenses@latest
```
Comment (mmatczuk) [inline Makefile:17]: Move to GitHub ones.

Actually we could move `golang.org/x/tools` to to top and separate the groups.

**Comment by mmatczuk** [comment]
Sweet. Let's add a makefile target to rerun it.

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -40,3 +40,6 @@ jobs:
 
       - name: Run e2e test
         run: cd e2e && ./run.sh
+
+      - name: verify third party licenses
```
Comment (mmatczuk) [inline .github/workflows/go.yml:44]: This is only relevant in release mode.

**Comment by mmatczuk** [comment]
Could we use just `verify-licenses` for brevity?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -35,6 +35,9 @@ jobs:
         with:
           go-version: "${{env.GO_VERSION}}"
 
+      - name: Verify licenses
```
Comment (mmatczuk) [inline .github/workflows/release.yml:38]: Please split to 2 steps to eliminate &&

File: Makefile [hunk 4]
```diff
@@ -10,15 +10,28 @@ include .version
...
+	go install github.com/goreleaser/goreleaser@$(GORELEASER_VERSION)
+	go install github.com/google/go-licenses@latest
+
 .PHONY: build
 build:
 	@goreleaser build --rm-dist --snapshot --single-target --skip-validate
 
+.PHONY: gen-licenses
+gen-licenses:
+	./bin/go-licenses report . --template third_party_license.tpl > third_party_licenses.txt
```
Comment (mmatczuk) [inline Makefile:26]: ./bin/ is not needed and it may be wrong if GOPATH is different.

Comment (Choraden) [reply Makefile]: right

**Review by mmatczuk — APPROVED** [review]

**Pull request merged**